From 206fa7f5c1279eeac0c6b3f7cb7c277ac7e80521 Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Mon, 9 Mar 2020 03:38:24 +0000
Subject: [PATCH 1/2] tiffgt: Link winmm if windows

fixes linking errors with libglut, although winmm is inside glut.pc,
FindGLUT.cmake provided by cmake does not check the pkg-config file
libglut.a(fg_init_mswin.c.obj):fg_init_mswin.c:(.text+0x135): undefined reference to `__imp_timeBeginPeriod'
libglut.a(fg_init_mswin.c.obj):fg_init_mswin.c:(.text+0x2e5): undefined reference to `__imp_timeEndPeriod'
libglut.a(fg_joystick_mswin.c.obj):fg_joystick_mswin.c:(.text+0x1d9): undefined reference to `__imp_joyGetPosEx'
libglut.a(fg_joystick_mswin.c.obj):fg_joystick_mswin.c:(.text+0x3f5): undefined reference to `__imp_joyGetDevCapsA'
libglut.a(fg_main_mswin.c.obj):fg_main_mswin.c:(.text+0xe86): undefined reference to `__imp_timeGetTime'
libglut.a(fg_main_mswin.c.obj):fg_main_mswin.c:(.text+0xea6): undefined reference to `__imp_timeGetTime'
---
 tools/CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index 6104ccd6..ffafbbcc 100755
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -158,7 +158,9 @@ if(OPENGL_SUPPORT)
   set_target_properties(tiffgt PROPERTIES LINKER_LANGUAGE CXX)
   target_link_libraries(tiffgt PRIVATE tiff tiff_port)
   target_link_libraries(tiffgt PRIVATE OpenGL::GL GLUT::GLUT)
-
+  if(WIN32)
+    target_link_libraries(tiffgt PRIVATE winmm)
+  endif()
   install(TARGETS tiffgt
           RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}")
 endif()
-- 
2.47.1

